bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/handler/MikrotikBaseThingHandler.java

```diff
@@ -0,0 +1,226 @@
+ * whenever required.
+ *
+ * @author Oleg Vivtash - Initial contribution
+ *
+ *
+ * @param <C> config - the config class used by this base thing handler
+ *
+ */
+@NonNullByDefault
+public abstract class MikrotikBaseThingHandler<C extends ConfigValidation> extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(MikrotikBaseThingHandler.class);
+    protected @Nullable C config;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    protected LocalDateTime lastModelsRefresh = LocalDateTime.now();
+    protected Map<String, State> currentState = new HashMap<>();
+
+    // public static boolean supportsThingType(ThingTypeUID thingTypeUID) <- in subclasses
+
+    public MikrotikBaseThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    protected @Nullable MikrotikRouterosBridgeHandler getVerifiedBridgeHandler() {
+        @Nullable
+        Bridge bridgeRef = getBridge();
+        if (bridgeRef != null && bridgeRef.getHandler() != null
+                && (bridgeRef.getHandler() instanceof MikrotikRouterosBridgeHandler)) {
+            return (MikrotikRouterosBridgeHandler) bridgeRef.getHandler();
+        }
+        return null;
+    }
+
+    protected final @Nullable RouterosDevice getRouteros() {
+        @Nullable
+        MikrotikRouterosBridgeHandler bridgeHandler = getVerifiedBridgeHandler();
+        return bridgeHandler == null ? null : bridgeHandler.getRouteros();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Handling command = {} for channel = {}", command, channelUID);
+        if (getThing().getStatus() == ONLINE) {
+            RouterosDevice routeros = getRouteros();
+            if (routeros != null) {
+                if (command == REFRESH) {
+                    throttledRefreshModels();
+                    refreshChannel(channelUID);
+                } else {
+                    try {
+                        executeCommand(channelUID, command);
+                    } catch (Exception e) {
+                        logger.warn("Unexpected error handling command = {} for channel = {} : {}", command, channelUID,
+                                e.getMessage());
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing {}!", getThing().getUID());
+        cancelRefreshJob();
+        if (getVerifiedBridgeHandler() == null) {
+            updateStatus(OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "This thing requires a RouterOS bridge");
+            return;
+        }
+
+        Class<?> klass = (Class<?>) (((ParameterizedType) getClass().getGenericSuperclass())
+                .getActualTypeArguments()[0]);
+        this.config = (C) getConfigAs(klass);
+        logger.trace("Config for for {} ({}) is {}", getThing().getUID(), getThing().getStatus(), config);
+
+        if (!config.isValid()) {
```

Comment:
The field can be null at this point, because it isn't inittialized in the constructor. So, you need to check if it is not null before accessing it: Example: if (job != null). But this would result in a compiler warning, as it is a field and could theoretically be modified by another thread exactly between the null check and the access in the next line. To mitigate this, you can store the field to a local variable before checking it:
```
Future<?> localPollingJob = pollingJob;
if (localPollingJob != null) {
    localPollingJob.cancel(true);
}
```

Please check all.

Reply:
Done

Reply:
To get rid of the warning, you need to add a null check to `localConfig` and use this instead of the field.